test/unit/org/apache/cassandra/utils/JVMStabilityInspectorTest.java

```diff
@@ -60,6 +62,14 @@ public void testKill() throws Exception
             JVMStabilityInspector.inspectThrowable(new FSReadError(new IOException(), "blah"));
             assertTrue(killerForTests.wasKilled());
 
+            killerForTests.reset();
+            JVMStabilityInspector.inspectThrowable(new FSWriteError(new IOException(), "blah"));
+            assertTrue(killerForTests.wasKilled());
+
+            killerForTests.reset();
+            JVMStabilityInspector.inspectThrowable(new CorruptSSTableException(new IOException(), "blah"));
```

Comment:
can you also add a second one where the corrupt sstable is wrapped in a runnable?